File: prodi.py
class Prodi :
    #access modifier private
    __nama_prodi = "";
    __kode = "";
    __course = "";

    # constructor
    def __init__(self, nama_prodi = "", kode = "", course = ""):
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__course = course

    # penggunaan getter dan setter
    def get_nama_prodi(self):
        return self.__nama_prodi
    
    def set_nama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi

    def get_kode(self):
        return self.__kode
    
    def set_kode(self, kode):
        self.__kode = kode
    
    def get_course(self):
        return self.__course
    
    def set_course(self, course):
        self.__course = course

    #fungsi untuk menambahkan isi array, note: tapi tidak terpakai
    def __str__(self):
        lines = []
        lines.append(f'{self.nama_prodi}, {self.kode}, {self.course}')
        return '\n'.join(lines)

File: cpp/prodi.cpp
//import library
#include <iostream>
#include <string>
#include <bits/stdc++.h>
//using standard namespace
using namespace std;
//Prodi sebagai anak yang diwariskan dari Mahasiswa
class Prodi
{
    //access modifier private
    private:
        string nama_prodi;
        string kode;
        string course;

    //access modifier publik    
    public:
        //constructor
        Prodi()
        {
            this->nama_prodi = " ";
            this->kode = " ";
            this->course = " ";
        }

        //penggunaan getter dan setter
        string get_nama_prodi()
        {
            return this->nama_prodi;
        }

        void set_nama_prodi(string nama_prodi)
        {
            this->nama_prodi = nama_prodi;
        }
        
        string get_kode()
        {
            return this->kode;
        }

        void set_kode(string kode)
        {
            this->kode = kode;
        }

        char get_course()
        {
            return this->course;
        }

        void set_course(string course)
        {
            this->course = course;
        }

        //destructor
        ~Prodi()
        {

        }
};